optgen/cmd/support/agg_gen.go

```diff
@@ -92,27 +110,45 @@ func (g *AggGen) genAggStringer(define AggDef) {
 		sqlName = define.SqlName
 	}
 	fmt.Fprintf(g.w, "func (a *%s)  String() string {\n", define.Name)
-	fmt.Fprintf(g.w, "    return fmt.Sprintf(\"%s(%%s)\", a.Child)\n", strings.ToUpper(sqlName))
+	if define.IsNary {
+		fmt.Fprintf(g.w, "    return fmt.Sprintf(\"%s(%%s)\", a.ChildExpressions)\n", strings.ToUpper(sqlName))
+	} else {
+		fmt.Fprintf(g.w, "    return fmt.Sprintf(\"%s(%%s)\", a.Child)\n", strings.ToUpper(sqlName))
+	}
 	fmt.Fprintf(g.w, "}\n\n")
 }
 
 func (g *AggGen) genAggWithChildren(define AggDef) {
 	fmt.Fprintf(g.w, "func (a *%s) WithChildren(children ...sql.Expression) (sql.Expression, error) {\n", define.Name)
-	fmt.Fprintf(g.w, "    res, err := a.unaryAggBase.WithChildren(children...)\n")
-	fmt.Fprintf(g.w, "    return &%s{unaryAggBase: *res.(*unaryAggBase)}, err\n", define.Name)
+	if define.IsNary {
+		fmt.Fprintf(g.w, "    res, err := a.naryAggBase.WithChildren(children...)\n")
```

Comment:
You might be able to clean up some of this forked code if you don't directly reference `naryAggBase` and just rely on Go to pass through calls to the embedded type of either naryAggBase or unaryAggBase. i.e. just outputting the code:
`res, err := a.WithChildren(children...)\n`

That could help reduce duplication and make this code easier to read.